io_scene_niftools/modules/nif_export/property/object/__init__.py

```diff
@@ -226,10 +226,10 @@ def export_inventory_marker(n_root, b_obj):
         if niftools_scene.game in ('SKYRIM',) and bs_inv_store:
             bs_inv = bs_inv_store[0]
             n_bs_inv_marker = NifClasses.BSInvMarker(n_root.context)
-            n_bs_inv_marker.name = bs_inv.name.encode()
-            n_bs_inv_marker.rotation_x = (-bs_inv.x % (2 * pi)) * 1000
-            n_bs_inv_marker.rotation_y = (-bs_inv.y % (2 * pi)) * 1000
-            n_bs_inv_marker.rotation_z = (-bs_inv.z % (2 * pi)) * 1000
+            n_bs_inv_marker.name = bs_inv.name
+            n_bs_inv_marker.rotation_x = round((-bs_inv.x % (2 * pi)) * 1000)
```

Comment:
This might be a good use case for a special subtype of ushort that is treated like a float?

Reply:
Could be. I don't know that it would necessarily be worth implementing if it was just this place, but bhkCMSChunk also uses a similar format (float value * 1000, rather than float value between 0 and 1 or -1 and 1). That being said, you'd still need the processing here to keep it within range, because the 2 * pi modulo is only valid for rotations, and we don't really do any processing of bhkCMSChunk anyways. Do you know of any other places with a similar format?

Reply:
True about the modulo. Not sure why it's set to negative, though.

ZT2's BF animation format (essentially a stripped down and compressed KF) uses the same encoding:
https://github.com/HENDRIX-ZT2/bfb-blender/blob/blender3/source/formats/bf/bf.xml

Reply:
The negative is because Blender's rotations are reversed to Skyrim's.
I also just noticed the Orientation field in FurniturePosition might be the same.
I assume you're referring to ushort1000 type. If both bf and nif use one, should it be in the base format? Or just keep it separate? I was moreso asking about other places in the nif xml, since I wasn't sure two fields warranted a completely new basic type.

Reply:
Not aware of any other occurences in standard NIF. I don't think those few occurences justify putting it in base/basic